src/screens/TopicPageScreen/index.js

```diff
@@ -447,13 +459,27 @@ const TopicPageScreen = (props) => {
           contentInsetAdjustmentBehavior={feeds?.length > 1 ? 'automatic' : 'never'}
         />
       </View>
-      <ButtonAddPostTopic topicName={topicName} onRefresh={onRefresh} />
+      <ButtonAddPostTopic topicName={topicName} onRefresh={onRefresh} followType={followType} />
       <BlockComponent
         ref={refBlockComponent}
         refresh={onBlockCompleted}
         refreshAnonymous={onDeleteBlockedPostCompleted}
         screen="topic_screen"
       />
+      <BottomSheetFollow
+        ref={bottomSheetFollowRef}
+        onClose={() => bottomSheetFollowRef.current.close()}
+        topicName={topicDetail.name}
+        memberCount={memberCount}
+        setMemberCount={setMemberCount}
+        isFollow={isFollow}
+        setIsFollow={setIsFollow}
+        followType={followType}
+        setFollowType={setFollowType}
+        getTopicDetail={getTopicDetail}
+        username={user?.myProfile.username}
+        profilePicture={user?.myProfile.profile_pic_path}
+      />
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of `ButtonAddPostTopic`, `BlockComponent`, and `BottomSheetFollow` components with their respective props is a significant change. These components are likely used to add posts, block certain actions, and handle follow actions respectively.
```diff
+ <ButtonAddPostTopic topicName={topicName} onRefresh={onRefresh} followType={followType} />
+ <BlockComponent
+   ref={refBlockComponent}
+   refresh={onBlockCompleted}
+   refreshAnonymous={onDeleteBlockedPostCompleted}
+   screen="topic_screen"
+ />
+ <BottomSheetFollow
+   ref={bottomSheetFollowRef}
+   onClose={() => bottomSheetFollowRef.current.close()}
+   topicName={topicDetail.name}
+   memberCount={memberCount}
+   setMemberCount={setMemberCount}
+   isFollow={isFollow}
+   setIsFollow={setIsFollow}
+   followType={followType}
+   setFollowType={setFollowType}
+   getTopicDetail={getTopicDetail}
+   username={user?.myProfile.username}
+   profilePicture={user?.myProfile.profile_pic_path}
+ />
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->